### PR TITLE
Include specifics of how reproducibility practices were implemented

### LINKED ISSUE #28: Mention how practice affects top and bottom 10 percent of students
> Finally, turning to the ‘Lots of practice’ section, I found the discussion very interesting, although if the paper needs to be trimmed, perhaps these pages – 12-13 – would be place to do it by condensing some of the discussion. More generally, I am wondering how this implementation accounts for the different abilities in the top 10 per cent of the class compared with the bottom 10 per cent or even median. How do the authors engage those different students, with their different needs, when it comes to practice?

I don't think this is anything we have data on, is it?

### LINKED ISSUE #31: Contrast course specific and program wide changes
> I think it would help to elaborate on the relationship between course specific changes and program wide changes. In the “Example lesson(s) of lots of practice” section (pp. 14-15), the authors outline how the principles of reproducibility have been fully integrated across courses in the Master of Data Science program, which strongly incentivizes students to practice what they’re taught over the duration of the program. Is it the case that these changes were adopted across the program simultaneously? Or were changes implemented in select courses initially and only more widely after some period of time? If it’s the latter, I would be very interested to know how students performed when the principles of reproducibility were not widely integrated throughout the curriculum. In other schools, it might be the case that there is limited appetite or support for a coordinated change across the curriculum. For readers in this situation, it would be beneficial to know more about the authors experiences implementing these changes only among select courses.

### PR DESCRIPTION
**Pull request by joelostblom**
- close #31
- close #28

### CODE REVIEW

File: manuscript/teaching-reproducibility.Rmd [hunk 1]
```diff
@@ -141,7 +141,7 @@ in classrooms of 10-200 learners from varying educational backgrounds,
 including undergraduate students taking their first data science course
 and graduate students looking to specialize in data science.
 We teach the plurality of our courses in the Master of Data Science program,
-which consists of 24 one-credit courses
+which consists of 24 one-credit courses over an eight month period
```
Comment (joelostblom) [inline manuscript/teaching-reproducibility.Rmd:144]: Added this here and removed the equivalent sentence in the reproducibility section.

File: manuscript/teaching-reproducibility.Rmd [hunk 2]
```diff
@@ -141,7 +141,7 @@ in classrooms of 10-200 learners from varying educational backgrounds,
...
 culminating in a six-credit-capstone course.
 As a pre-requisite for this program
 students are generally required to have taken one course each
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd:147]: On the line below it says stat and prob, but it should be stat or prob

Comment (ttimbers) [reply manuscript/teaching-reproducibility.Rmd:147]: Change lines below from:

```
Typically,
students also have a few years of working experience (or equivalent),
but our cohorts includes both recent graduates from Bachelor's degrees
as well as learners from senior company positions
looking to bend their career trajectories towards data science.
```

to:

```
Most of these graduate students have several years of professional work experience (or equivalent),
however our cohorts also include both recent graduates from Bachelor's degrees
as well as learners from senior company positions
looking to bend their career trajectories towards data science.
```

File: manuscript/teaching-reproducibility.Rmd [hunk 3]
```diff
@@ -729,19 +729,39 @@ Additionally, we interleave these practices as
...
 This latter strategy parallels how they will employ reproducibility 
 habits later in their carers
-(a detailed example follows in the next section).
-The tools we chose to teach with are data science reproducibility software
-that students are likely to employ in workflows in their future work.
-This sustained practice not only enforces students' habits,
-but also increases their proficiency using authentic reproducibility software
-and when they into problems, it is in an environment
-where they can easily reach out for help without feeling intimidated to ask.
+(a detailed example follows in the next section),
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd]:
```suggestion
(a detailed example follows in the next section).
```

File: manuscript/teaching-reproducibility.Rmd [hunk 4]
```diff
@@ -729,19 +729,39 @@ Additionally, we interleave these practices as
...
+and the skill bar is put at a level
+that we require all student to meet
+after introduced to version control topic
+during the first few weeks of class.
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd]: This should be it's own sentence. And it needs to be edited. I find it hard to parse and digest. Or we could remove it?

File: manuscript/teaching-reproducibility.Rmd [hunk 5]
```diff
@@ -729,19 +729,39 @@ Additionally, we interleave these practices as
 grade while completing assignments focused on other data science topics.
 This latter strategy parallels how they will employ reproducibility 
 habits later in their carers
-(a detailed example follows in the next section).
-The tools we chose to teach with are data science reproducibility software
-that students are likely to employ in workflows in their future work.
-This sustained practice not only enforces students' habits,
-but also increases their proficiency using authentic reproducibility software
-and when they into problems, it is in an environment
-where they can easily reach out for help without feeling intimidated to ask.
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd:738]: Why was this removed? I like it. If there is good reason to remove it however, please let me know and we can do so.

File: manuscript/teaching-reproducibility.Rmd [hunk 6]
```diff
@@ -729,19 +729,39 @@ Additionally, we interleave these practices as
...
+during the first few weeks of class.
+To engage students at different levels of skill
+and interest in reproducibility,
+we have more difficult exercises
+in the courses specifically teaching these topics
+(including demanding optional exercises for bonus points).
+Students less experienced and interested in these topics
+have shared with us that they find the worksheets
+really helpful in preparing them
+to successfully complete the lab questions.
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd]:
```suggestion
We have observed a wide-range of interest, experience 
and skill-level in our students when teaching reproducibility.
Feedback from instructor and student-led surveys have indicated that
the guided electronic worksheets with automated feedback
have increased students perceived learning.
We hypothesize this is particularly true for students' who
have less interest, experience or skills in the area of reproducibility.
To challenge students who have more interest, experience or skills in the area of reproducibility,
we routinely include demanding optional questions in our homework assignments 
that allow these students to engage more deeply with the material for a small amount of bonus points.
We hope these strategies help to engage students across the wide range of interest, experience 
and skills in the area of reproducibility that we see in our classrooms.
```

File: manuscript/teaching-reproducibility.Rmd [hunk 7]
```diff
@@ -729,19 +729,39 @@ Additionally, we interleave these practices as
...
+was implemented across the Master of Data Science program from year one.
+To make this feasible,
+postdoctoral teaching and learning fellows were hired
+to help faculty implement this in their courses
+even if those faculty themselves were not familiar with version control.
+The teaching team model in the Master of Data Science program
+greatly facilitates us doing program-wide changes like this.
+This stems from our shared vision for the program
+(which was drafted collaboratively by the team),
+our once-per term academic retreats where we reflect on how things went,
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd]:
```suggestion
our once-per term academic retreats where we reflect on how things went in each course,
```

File: manuscript/teaching-reproducibility.Rmd [hunk 8]
```diff
@@ -758,7 +778,7 @@ but there are other tools that can do this,
 including [GitHub Classroom](https://classroom.github.com/) 
 and [Classy](https://github.com/ubccpsc/classy).
 
-To incentivize doing this using Git 
+To incentivize doing students to submit their assignment by using Git
```
Comment (ttimbers) [inline manuscript/teaching-reproducibility.Rmd]:
```suggestion
To incentivize students to submit their assignment using Git 
```

File: manuscript/teaching-reproducibility.Rmd (hunk 5 above)
Comment (ttimbers) [reply manuscript/teaching-reproducibility.Rmd:738]: Ohhhhh, now I see, this was just moved below! You can ignore the above comment.

**Review by ttimbers — COMMENTED** [review]
Some minor suggested changes
